dynamic/clientset/clientset.go

```diff
@@ -71,85 +70,82 @@ func (cs *Clientset) Kind(apiVersion, kind string) (*ResourceClient, error) {
 }
 
 func (cs *Clientset) resource(apiResource *dynamicdiscovery.APIResource) *ResourceClient {
+	client := cs.dc.Resource(apiResource.GroupVersionResource())
 	return &ResourceClient{
-		NamespaceableResourceInterface: cs.dc.Resource(apiResource.GroupVersionResource()),
-		resource:                       apiResource,
+		ResourceInterface: client,
+		APIResource:       apiResource,
+		rootClient:        client,
 	}
 }
 
+// ResourceClient is a combination of APIResource and a dynamic Client.
+//
+// Passing this around makes it easier to write code that deals with arbitrary
+// resource types and often needs to know the API discovery details.
+// This wrapper also adds convenience functions that are useful for any client.
+//
+// It can be used on either namespaced or cluster-scoped resources.
+// Call Namespace() to return a client that's scoped down to a given namespace.
 type ResourceClient struct {
-	dynamic.NamespaceableResourceInterface
-	resource *dynamicdiscovery.APIResource
-}
-
-type NamespacedResourceClient struct {
 	dynamic.ResourceInterface
-	resource *dynamicdiscovery.APIResource
-}
-
-func (rc *ResourceClient) Namespace(namespace string) *NamespacedResourceClient {
-	return &NamespacedResourceClient{
-		ResourceInterface: rc.NamespaceableResourceInterface.Namespace(namespace),
-		resource:          rc.resource,
-	}
-}
+	*dynamicdiscovery.APIResource
 
-func (rc *ResourceClient) Kind() string {
-	return rc.resource.Kind
+	rootClient dynamic.NamespaceableResourceInterface
 }
 
-func (rc *ResourceClient) GroupVersion() schema.GroupVersion {
-	return rc.resource.GroupVersion()
-}
-
-func (rc *ResourceClient) GroupResource() schema.GroupResource {
-	return schema.GroupResource{
-		Group:    rc.resource.GroupVersion().Group,
-		Resource: rc.resource.Name,
+// Namespace returns a copy of the ResourceClient with the client namespace set.
+//
+// This can be chained to set the namespace to something else.
```

Comment:
Does it mean that `.Namespace("a").Namespace("b")` will have the namespace set as `b`?

Reply:
Yes